internal/controllers/addon_utils.go

```diff
@@ -64,6 +64,7 @@ func (r *AddonReconciler) reportTerminationStatus(
 // Report Addon status to communicate that the resource is misconfigured
 func (r *AddonReconciler) reportConfigurationError(
 	ctx context.Context, addon *addonsv1alpha1.Addon, message string) error {
+	// TODO: remove the following 2 lines of code
```

Comment:
Why? Could you please also add the reason for the same? :)

Reply:
Coz line 76-77 are already accomplishing the update in `ObservedGeneration` and `Phase` just like in `reportTerminationStatus` and `reportReadinessState`

Reply:
Ah, okay. Missed that :)